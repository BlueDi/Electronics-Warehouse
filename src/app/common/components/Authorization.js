import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { withCookies } from 'react-cookie';

class Authorization extends Component {
  checkAuth(securityLevel) {
    return securityLevel > 1;
  }

  render() {
    var isAuth = this.checkAuth(this.props.cookies.get('security'));
    return isAuth ? this.props.children : <Redirect to="/" />;
  }
}

export default withCookies(Authorization);
