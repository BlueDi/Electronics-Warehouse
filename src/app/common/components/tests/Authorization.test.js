import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Authorization from '../Authorization';

test('Authorization Render', () => {
  const component = renderer.create(
    <Router>
      <Authorization
        param={'cenas'}
        cookies={{
          cenas: 'true',
          getAll: () => {},
          get: () => {
            return true;
          },
          addChangeListener: () => {}
        }}
      />
    </Router>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
