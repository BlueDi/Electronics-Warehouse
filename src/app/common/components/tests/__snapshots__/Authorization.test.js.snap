// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorization Render 1`] = `null`;
